libs/common/src/types/selected-fee-records-details.ts

```diff
@@ -1,6 +1,12 @@
 import { CurrencyAndAmount } from './currency';
 import { ReportPeriod } from './utilisation-reports';
 
+export type SelectedFeeRecordsPaymentDetails = {
+  value: CurrencyAndAmount;
```

Comment:
Why don't we store the `currency` and `amount` at the top level instead of inside this `value` field?

Reply:
I think I can see why - it's so that you can call `getFormattedCurrencyAndAmount` directly on the `value` field, right?

Reply:
I think I just liked it, in my head they are very linked and so this just made sense. I do see now that I consider it more that perhaps it isn't quite as clear cut as I thought so I actually don't think it matters too much either way, but it does make it nice for the formatting as you point out.

Reply:
My main concern with the word `value` was semantic as it feels a bit strange to me to access the "amount" by using `details.value.amount`, but I also think it can make sense as is. With respect to formatting, I did a bit of experimenting and found that any type which overlaps with the `CurrencyAndAmount` type can be used in `getFormattedCurrencyAndAmount`. For example, the following is valid:

```typescript
type SomeDetails = {
  currency: Currency;
  amount: number;
  id: number;
  ... other fields
};

const details: SomeDetails = {
  currency: 'GBP',
  amount: 100,
  id: 1,
  ...
};

const formattedCurrencyAndAmount = getFormattedCurrencyAndAmount(someDetails); // works fine
```

Just to try and avoid making things more confusing that they already are with all these fee record and payment types, I think it may be better to keep the `currency` and `amount` fields at the top level of this object, but I'm not sure whether or not it does actually make things clearer